tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/PrunedPartitionBuilder.java

```diff
@@ -189,19 +211,29 @@ private static boolean canBePruned(TiTableInfo tblInfo, Expression filter) {
     TiPartitionInfo partInfo = tableInfo.getPartitionInfo();
     List<Expression> partExprs = new ArrayList<>();
     TiParser parser = new TiParser(tableInfo);
+    // check year expression
+    // rewrite filter condition
+    // purchased > '1995-10-10'
+    // year(purchased) > year('1995-10-10')
+    // evaled_purchased > 1995
     String partExprStr = tableInfo.getPartitionInfo().getExpr();
     partExpr = parser.parseExpression(partExprStr);
+    partExprColRefs = PredicateUtils.extractColumnRefFromExpression(partExpr);
     // when it is not range column case, only first element stores useful info.
     for (int i = 0; i < partInfo.getDefs().size(); i++) {
       TiPartitionDef pDef = partInfo.getDefs().get(i);
+      String current = pDef.getLessThan().get(0);
+      String leftHand = "";
+      if (current.equals("MAXVALUE")) {
+        leftHand = "true";
```

Comment:
I can understand reconstruct partition range expression, but what is "true"? And is that how you set a placeholder for positive INF?

Reply:
Yes.